### PR TITLE
All unit tests running using GitHub actions

### PR DESCRIPTION
**Pull request by tankefugl**
This pull request enables all unit tests running in GitHub actions. It also adds the scaffolding needed to add caching to the docker image building, which needs a docker repository account to be set up completely.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,4 +35,4 @@ WORKDIR /silcam
 RUN python setup.py develop
 
 # Run the PySilCam tests as default entrypoint
-ENTRYPOINT ["python", "setup.py", "test"]
+CMD ["python", "setup.py", "test"]
```
Comment (tankefugl) [inline Dockerfile:38]: Changed from ENTRYPOINT to CMD to allow the user of the container to override it. Makes it a bit easier to investigate issues locally.

File: README.md [hunk 2]
```diff
@@ -111,4 +111,4 @@ If you want to mount a local folder inside the docker environment, use -v [local
 License
 -------
 
-BSD3 license. See LICENSE
+PySilCam is licensed under the BSD3 license. See LICENSE
```
Comment (tankefugl) [inline README.md:114]: Small addition and just an autoformat (removing trailing spaces) in this file.

File: .github/workflows/docker-build-and-test.yml [hunk 3]
```diff
@@ -1,20 +1,39 @@
 name: Docker build and test
 
-on:
-  push:
-    branches: [ public ]
-  pull_request:
-    branches: [ public ]
+on: [push, pull_request]
```
Comment (tankefugl) [inline .github/workflows/docker-build-and-test.yml:3]: Will now trigger on all pushes and pull requests.

**Review by emlynjdavies — APPROVED** [review]

**Comment by emlynjdavies** [comment]
code changes look logical, but tests are failing....

**Comment by tankefugl** [comment]
Closing this, taking it to the .gitignore PR.

**Pull request closed without merging**